src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -65,7 +65,7 @@ namespace ErrorCodes
 class FunctionDictHelper : WithContext
 {
 public:
-    explicit FunctionDictHelper(ContextPtr context_) : WithContext(context_) {}
+    explicit FunctionDictHelper(ContextPtr context_) : WithContext(context_->getGlobalContext()) {}
```

Comment:
There's current_context->checkAccess() on line 77 below. Will it work correctly with global context, or does it take the current user's identity from the non-global part of context?

Reply:
Most likely not :(
Let's add a test case.

PS. There is also a session context.